Intermittent test failure in assumptions

This test fails intermittently on Python 3.5 or 3.6:
```console
$ pytest sympy/core/tests/test_arit.py -k test_Add_Mul_is_finite
==================================================================== test session starts =====================================================================
platform darwin -- Python 3.5.7, pytest-4.3.1, py-1.8.0, pluggy-0.9.0
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/enojb/current/sympy/sympy/.hypothesis/examples')
architecture: 64-bit
cache:        yes
ground types: gmpy 1.17

rootdir: /Users/enojb/current/sympy/sympy, inifile: pytest.ini
plugins: xdist-1.27.0, instafail-0.4.1, forked-1.0.2, doctestplus-0.3.0, cov-2.7.1, hypothesis-4.32.3
collected 88 items / 87 deselected / 1 selected                                                                                                              

sympy/core/tests/test_arit.py F                                                                                                                        [100%]

========================================================================== FAILURES ==========================================================================
___________________________________________________________________ test_Add_Mul_is_finite ___________________________________________________________________

    def test_Add_Mul_is_finite():
        x = Symbol('x', extended_real=True, finite=False)
    
        assert sin(x).is_finite is True
        assert (x*sin(x)).is_finite is None
        assert (x*atan(x)).is_finite is False
        assert (1024*sin(x)).is_finite is True
>       assert (sin(x)*exp(x)).is_finite is None
E       assert False is None
E        +  where False = (sin(x) * exp(x)).is_finite
E        +    where sin(x) = sin(x)
E        +    and   exp(x) = exp(x)

sympy/core/tests/test_arit.py:390: AssertionError
```

--- a/sympy/functions/elementary/exponential.py
+++ b/sympy/functions/elementary/exponential.py
@@ -6,7 +6,7 @@ from sympy.core.cache import cacheit
 from sympy.core.compatibility import range
 from sympy.core.function import (Function, ArgumentIndexError, _coeff_isneg,
         expand_mul)
-from sympy.core.logic import fuzzy_and, fuzzy_not
+from sympy.core.logic import fuzzy_and, fuzzy_not, fuzzy_or
 from sympy.core.mul import Mul
 from sympy.core.numbers import Integer, Rational
 from sympy.core.power import Pow
@@ -402,7 +402,10 @@ class exp(ExpBase):
             return arg2.is_even
 
     def _eval_is_complex(self):
-        return self.args[0].is_complex
+        def complex_extended_negative(arg):
+            yield arg.is_complex
+            yield arg.is_extended_negative
+        return fuzzy_or(complex_extended_negative(self.args[0]))
 
     def _eval_is_algebraic(self):
         s = self.func(*self.args)

--- a/sympy/functions/elementary/tests/test_exponential.py
+++ b/sympy/functions/elementary/tests/test_exponential.py
@@ -108,6 +108,8 @@ def test_exp_infinity():
     assert refine(exp(-I*oo)) is nan
     assert exp(y*I*oo) != nan
     assert exp(zoo) is nan
+    x = Symbol('x', extended_real=True, finite=False)
+    assert exp(x).is_complex is None
 
 
 def test_exp_subs():

